cmd/aws-k8s-agent/main.go

```diff
@@ -59,6 +60,8 @@ func _main() int {
 	ipamContext, err := ipamd.New(rawK8SClient, cacheK8SClient)
 	if err != nil {
 		log.Errorf("Initialization failure: %v", err)
+		// Allow time for event to be broadcast in case of missing permissions
```

Comment:
Curious, why do we need this? Do we see a pod tear down in this case? Isn't this just a container restart case? and so even if the event broadcast takes a while, it should eventually show up on Pod events. On top of it, we retry on ec2 call failures and that should give us ample time. I'm assuming this is here based on what you observed during testing but trying to understand more on why this is required.

Reply:
That's right, I observed that events take a while to show up. If the container restarts before that, the events were getting lost. 
I suspect this is because we start a new broadcaster on each restart, so this might still be required. Will look into this a bit more & update.

Reply:
If this is needed, let's move it to the `BroadcastEvent` function instead. We can land here for a variety of reasons and this is only required for a particular scenario this PR is addressing.

Reply:
sleep statement is not required here as the BroadcastEvent call is moved to awsutils.